### PR TITLE
Refactors closeNote to Redux

### PR DESCRIPTION
**Pull request by belcherj**
### Fix

Refactors `closeNote` moving the function from the flux actions to Redux. In doing this it required the removal of the note list open/closed that was being stored in the component state of app.tsx.

### Test
1. Have SImplenote open with a small width so only the note list appears
2. Click a note
3. Click the back button at the top
4. Click note again
5. Trash it, does note list appear?
6. Open Trashed note list
7. Click into note
8. Click back button at top
9. Permanently trash note, should go back to trashed note list
10. Empty trash, should see blank note editor
11. Go back to all notes
12. Create new note
13. Play with search

### CODE REVIEW

File: lib/state/ui/reducer.ts [hunk 1]
```diff
@@ -52,24 +51,48 @@ const simperiumConnected: A.Reducer<boolean> = (state = false, action) =>
...
+        newSet.delete('noteList');
+        return newSet;
+      }
+    case 'TAG_DRAWER_TOGGLE': {
+      if (action.show) {
+        return new Set(state).add('tagDrawer');
+      } else {
+        const newSet = new Set(state);
+        newSet.delete('tagDrawer');
+        return newSet;
```
Comment (dmsnell) [inline lib/state/ui/reducer.ts]: as noted in Slack, we're going to remove `visiblePanes` right?

Comment (belcherj) [reply lib/state/ui/reducer.ts]: yes, I hadn't updated yet.

File: lib/note-toolbar/index.tsx [hunk 2]
```diff
@@ -180,10 +180,10 @@ const mapStateToProps: S.MapState<StateProps> = ({
...
-const mapDispatchToProps: S.MapDispatch<DispatchProps> = dispatch => ({
-  closeNote: () => dispatch(closeNote()),
-  toggleEditMode: () => dispatch(toggleEditMode()),
-  toggleNoteInfo: () => dispatch(toggleNoteInfo()),
-});
+const mapDispatchToProps: S.MapDispatch<DispatchProps> = {
+  closeNote,
+  toggleEditMode,
+  toggleNoteInfo,
+};
```
Comment (dmsnell) [inline lib/note-toolbar/index.tsx]: 😎

File: lib/app.tsx [hunk 3]
```diff
@@ -217,25 +216,11 @@ export const App = connect(
...
-      }
-
-      if (isSmallScreen !== prevProps.isSmallScreen) {
-        this.setState({
-          isNoteOpen: Boolean(
-            this.props.appState.note &&
-              (settings.focusModeEnabled || !isSmallScreen)
-          ),
-        });
-      }
```
Comment (dmsnell) [inline lib/app.tsx]: Earmarking to test…

File: lib/state/ui/reducer.ts [hunk 4]
```diff
@@ -39,6 +37,19 @@ const listTitle: A.Reducer<T.TranslatableString> = (
...
+  switch (action.type) {
+    case 'CLOSE_NOTE': {
+      return true;
+    }
+    case 'App.selectNote':
+      return false;
+
+    default:
+      return state;
+  }
```
Comment (dmsnell) [inline lib/state/ui/reducer.ts]: For actions like `deleteNoteForever` and `restoreNote` and the like is this now handled automatically by some other code, that the note closes?

File: lib/state/ui/reducer.ts [hunk 5]
```diff
@@ -52,19 +63,6 @@ const simperiumConnected: A.Reducer<boolean> = (state = false, action) =>
...
-  action
-) => {
-  if ('TAG_DRAWER_TOGGLE' === action.type) {
-    return action.show
-      ? union(state, ['tagDrawer'])
-      : difference(state, ['tagDrawer']);
-  }
-
-  return state;
-};
```
Comment (dmsnell) [inline lib/state/ui/reducer.ts]: Thanks for removing this. Probably would have been nice to do it in a separate PR with its own traceable history, or at least a mention in the `git` commit, since now we don't have any explanation for why this was removed in the code or `git` history - it just disappears.

File: lib/flux/app-state.ts [hunk 6]
```diff
@@ -395,12 +389,10 @@ export const actionMap = new ActionMap({
         note: T.NoteEntity;
         previousIndex: number;
       }) {
-        return dispatch => {
+        return () => {
           if (note) {
             note.data.deleted = true;
             noteBucket.update(note.id, note.data);
-
-            dispatch(this.action('closeNote', { previousIndex }));
```
Comment (dmsnell) [inline lib/flux/app-state.ts]: Are we removing the functionality that the `previousIndex` provided?

Comment (dmsnell) [reply lib/flux/app-state.ts]: Looks like this was broken in #1851 so it won't appear in testing of this PR

File: lib/state/ui/reducer.ts (hunk 4 above)
Comment (dmsnell) [reply lib/state/ui/reducer.ts]: Noted in testing: the note closes automatically due to the filtering middleware which implicitly unselects it _as it no longer exists_.
